Rollup snapshot error

Seen on 0.34, no additional details yet.

```
13:46:49.218 ERROR heduler-Concurrent-2 | i.h.HierarchicalTableImpl | HierarchicalTable.snapshot error at nodeId=4294967296: java.lang.IllegalArgumentException: size 512 is incompatible with capacity 488
	at io.deephaven.chunk.ChunkBase.internalSetSize(ChunkBase.java:48)
	at io.deephaven.chunk.WritableChunk.setSize(WritableChunk.java:75)
	at io.deephaven.engine.table.impl.sources.immutable.ImmutableConstantIntSource.fillChunk(ImmutableConstantIntSource.java:59)
	at io.deephaven.engine.table.impl.hierarchical.HierarchicalTableImpl$NodeFillContext.fillNormalDestinations(HierarchicalTableImpl.java:1492)
	at io.deephaven.engine.table.impl.hierarchical.HierarchicalTableImpl.consumeRowsUntilNextExpansion(HierarchicalTableImpl.java:1390)
	at io.deephaven.engine.table.impl.hierarchical.HierarchicalTableImpl.visitExpandedNode(HierarchicalTableImpl.java:1320)
	at io.deephaven.engine.table.impl.hierarchical.HierarchicalTableImpl.visitExpandedNode(HierarchicalTableImpl.java:1196)
	at io.deephaven.engine.table.impl.hierarchical.HierarchicalTableImpl.visitExpandedNode(HierarchicalTableImpl.java:1327)
	at io.deephaven.engine.table.impl.hierarchical.HierarchicalTableImpl.traverseExpansionsAndFillSnapshotChunks(HierarchicalTableImpl.java:1190)
	at io.deephaven.engine.table.impl.hierarchical.HierarchicalTableImpl.lambda$snapshotData$7(HierarchicalTableImpl.java:1156)
	at io.deephaven.engine.table.impl.remote.ConstructSnapshot.callDataSnapshotFunction(ConstructSnapshot.java:1262)
	at io.deephaven.engine.table.impl.remote.ConstructSnapshot.callDataSnapshotFunction(ConstructSnapshot.java:1168)
	at io.deephaven.engine.table.impl.hierarchical.HierarchicalTableImpl.snapshotData(HierarchicalTableImpl.java:1153)
	at io.deephaven.engine.table.impl.hierarchical.HierarchicalTableImpl.snapshot(HierarchicalTableImpl.java:959)
	at io.deephaven.engine.table.impl.hierarchical.RollupTableImpl.snapshot(RollupTableImpl.java:59)
	at io.deephaven.server.hierarchicaltable.HierarchicalTableViewSubscription.buildAndSendSnapshot(HierarchicalTableViewSubscription.java:321)
	at io.deephaven.server.hierarchicaltable.HierarchicalTableViewSubscription.process(HierarchicalTableViewSubscription.java:286)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at io.deephaven.server.runner.scheduler.SchedulerModule$ThreadFactory.lambda$newThread$0(SchedulerModule.java:100)
	at java.base/java.lang.Thread.run(Thread.java:1583)
13:46:49.219 ERROR heduler-Concurrent-2 | i.d.s.s.SessionService    | Internal Error '8ed4bb06-72da-4ae0-ae4b-f83c685fac8f' io.deephaven.engine.exceptions.SnapshotUnsuccessfulException: Failed to execute function concurrently despite consistent state
	at io.deephaven.engine.table.impl.remote.ConstructSnapshot.callDataSnapshotFunction(ConstructSnapshot.java:1340)
	at io.deephaven.engine.table.impl.remote.ConstructSnapshot.callDataSnapshotFunction(ConstructSnapshot.java:1168)
	at io.deephaven.engine.table.impl.hierarchical.HierarchicalTableImpl.snapshotData(HierarchicalTableImpl.java:1153)
	at io.deephaven.engine.table.impl.hierarchical.HierarchicalTableImpl.snapshot(HierarchicalTableImpl.java:959)
	at io.deephaven.engine.table.impl.hierarchical.RollupTableImpl.snapshot(RollupTableImpl.java:59)
	at io.deephaven.server.hierarchicaltable.HierarchicalTableViewSubscription.buildAndSendSnapshot(HierarchicalTableViewSubscription.java:321)
	at io.deephaven.server.hierarchicaltable.HierarchicalTableViewSubscription.process(HierarchicalTableViewSubscription.java:286)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at io.deephaven.server.runner.scheduler.SchedulerModule$ThreadFactory.lambda$newThread$0(SchedulerModule.java:100)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: io.deephaven.engine.table.impl.hierarchical.HierarchicalTableSnapshotException: HierarchicalTable.snapshot error
	at io.deephaven.engine.table.impl.hierarchical.HierarchicalTableImpl.visitExpandedNode(HierarchicalTableImpl.java:1344)
	at io.deephaven.engine.table.impl.hierarchical.HierarchicalTableImpl.visitExpandedNode(HierarchicalTableImpl.java:1196)
	at io.deephaven.engine.table.impl.hierarchical.HierarchicalTableImpl.visitExpandedNode(HierarchicalTableImpl.java:1327)
	at io.deephaven.engine.table.impl.hierarchical.HierarchicalTableImpl.traverseExpansionsAndFillSnapshotChunks(HierarchicalTableImpl.java:1190)
	at io.deephaven.engine.table.impl.hierarchical.HierarchicalTableImpl.lambda$snapshotData$7(HierarchicalTableImpl.java:1156)
	at io.deephaven.engine.table.impl.remote.ConstructSnapshot.callDataSnapshotFunction(ConstructSnapshot.java:1262)
	... 13 more
Caused by: java.lang.IllegalArgumentException: size 512 is incompatible with capacity 488
	at io.deephaven.chunk.ChunkBase.internalSetSize(ChunkBase.java:48)
	at io.deephaven.chunk.WritableChunk.setSize(WritableChunk.java:75)
	at io.deephaven.engine.table.impl.sources.immutable.ImmutableConstantIntSource.fillChunk(ImmutableConstantIntSource.java:59)
	at io.deephaven.engine.table.impl.hierarchical.HierarchicalTableImpl$NodeFillContext.fillNormalDestinations(HierarchicalTableImpl.java:1492)
	at io.deephaven.engine.table.impl.hierarchical.HierarchicalTableImpl.consumeRowsUntilNextExpansion(HierarchicalTableImpl.java:1390)
	at io.deephaven.engine.table.impl.hierarchical.HierarchicalTableImpl.visitExpandedNode(HierarchicalTableImpl.java:1320)
	... 18 more
```

--- a/engine/table/src/main/java/io/deephaven/engine/table/impl/hierarchical/HierarchicalTableImpl.java
+++ b/engine/table/src/main/java/io/deephaven/engine/table/impl/hierarchical/HierarchicalTableImpl.java
@@ -1473,10 +1473,12 @@ abstract class HierarchicalTableImpl<IFACE_TYPE extends HierarchicalTable<IFACE_
 
             try (final RowSequence.Iterator rowsIter =
                     rows.size() >= chunkSize ? rows.getRowSequenceIterator() : null) {
-                final RowSequence chunkRows =
-                        rowsIter == null ? rows : rowsIter.getNextRowSequenceWithLength(chunkSize);
-                final int chunkRowsSize = chunkRows.intSize();
                 do {
+                    final RowSequence chunkRows = rowsIter == null
+                            ? rows
+                            : rowsIter.getNextRowSequenceWithLength(chunkSize);
+                    final int chunkRowsSize = chunkRows.intSize();
+                    Assert.leq(chunkRowsSize, "chunkRowsSize", capacity, "capacity");
                     for (int di = 0, ci = columns.nextSetBit(0); ci >= 0; ++di, ci = columns.nextSetBit(ci + 1)) {
                         if (ci == ROW_EXPANDED_COLUMN_INDEX) {
                             continue;

